### PR TITLE
Support enhanced database reporting for gRPC

### PR DESCRIPTION
**Pull request by misterpoe**

### CODE REVIEW

**Comment by misterpoe** [comment]
@matthewloring @ofrobots @cristiancavalli PTAL

File: lib/hooks/userspace/hook-grpc.js [hunk 1]
```diff
@@ -25,6 +25,15 @@ var agent;
 
 var SUPPORTED_VERSIONS = '0.13 - 0.15';
 
+function findFirstIndex(arr, predicate) {
```
Comment (matthewloring) [inline lib/hooks/userspace/hook-grpc.js]: Let's bring in [lodash.find](https://lodash.com/docs#find) for this. If only [Array.prototype.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) was present in `0.12` 😿

File: test/hooks/test-trace-grpc.js [hunk 2]
```diff
@@ -199,69 +217,125 @@ Object.keys(versions).forEach(function(version) {
...
+      });
+    });
+
+    it('should trace metadata for server requests', function(done) {
+      common.runInTransaction(function(endTransaction) {
+        var stream = client.testServerStream({n: 42}, metadata);
+        stream.on('data', function(data) {});
+        stream.on('status', function(status) {
+          endTransaction();
+          assert.strictEqual(status.code, grpc.status.OK);
```
Comment (matthewloring) [inline test/hooks/test-trace-grpc.js]: Would it be possible to report this status code with `enhancedDatabaseReporting`?

File: lib/hooks/userspace/hook-grpc.js (hunk 1 above)
Comment (misterpoe) [inline lib/hooks/userspace/hook-grpc.js]: Done

File: test/hooks/test-trace-grpc.js (hunk 2 above)
Comment (misterpoe) [inline test/hooks/test-trace-grpc.js]: Yes, we can listen for the 'status' event. I've added the test for that to the standard server stream request above (this test is for metadata).

**Comment by matthewloring** [comment]
LGTM.

**Pull request merged**